File: NetCoreGameServer/NetCoreGameServer/Data/Model/ChatMessage.cs
namespace NetCoreGameServer.Data.Model;

public class ChatMessage
{
    public string From { get; set; }
    public string To { get; set; }
    public string Message { get; set; }
    public ChatMessageType Type { get; set; }
    public long Timestamp { get; set; } = DateTimeOffset.UtcNow.ToUnixTimeSeconds();
}

File: next/src/models/ChatMessage.ts
enum ChatMessageType {
  GlobalChat,
  GroupChat,
  PrivateChat,
  SystemMessage
}

export type ChatMessage = {
  from: string;
  to: string;
  message: string;
  type: ChatMessageType;
  timestamp: number;
};

export default ChatMessage;
export { ChatMessageType };